test/freeze.js

```diff
@@ -1,16 +1,24 @@
+/**
+ * @typedef {import('..').Plugin} Plugin
+ */
+
 import test from 'tape'
 import {SimpleCompiler, SimpleParser} from './util/simple.js'
 import {unified} from '../index.js'
 
 test('freeze()', (t) => {
-  const frozen = unified().use(config).freeze()
+  const frozen = unified()
+    .use(function () {
+      // Note: TS has a bug so setting `this.Parser` and such doesn’t work,
+      // but assigning is fine.
+      Object.assign(this, {
+        Parser: SimpleParser,
+        Compiler: SimpleCompiler
+      })
```

Comment:
This is super weird. I found this hard to believe, but you’re right. This works in TypeScript, but not in type checked JavaScript :thinking: